google/cloud/forseti/scanner/scanner.py

```diff
@@ -36,27 +93,39 @@ def run(model_name=None, progress_queue=None, service_config=None):
     global_configs = service_config.get_global_config()
     scanner_configs = service_config.get_scanner_config()
 
-    violation_access = scanner_dao.define_violation(service_config.engine)
-    service_config.violation_access = violation_access
-
-    runnable_scanners = scanner_builder.ScannerBuilder(
-        global_configs, scanner_configs, service_config, model_name,
-        None).build()
-
-    # pylint: disable=bare-except
-    for scanner in runnable_scanners:
-        try:
-            scanner.run()
-            progress_queue.put('Running {}...'.format(
-                scanner.__class__.__name__))
-        except:
-            log_message = 'Error running scanner: {}'.format(
-                scanner.__class__.__name__)
-            progress_queue.put(log_message)
-            LOGGER.error(log_message, exc_info=True)
-    # pylint: enable=bare-except
-    log_message = 'Scan completed!'
-    progress_queue.put(log_message)
-    progress_queue.put(None)
-    LOGGER.info(log_message)
-    return 0
+    with service_config.scoped_session() as session:
+        service_config.violation_access = scanner_dao.ViolationAccess(session)
+        model_description = (
+            service_config.model_manager.get_description(model_name))
+        inventory_index_id = (
+            model_description.get('source_info').get('inventory_index_id'))
+        scanner_index_id = init_scanner_index(session, inventory_index_id)
+        runnable_scanners = scanner_builder.ScannerBuilder(
+            global_configs, scanner_configs, service_config, model_name,
+            None).build()
+
+        # pylint: disable=bare-except
+        succeeded = []
+        failed = []
```

Comment:
I mentioned this before, instead of having 2 lists here, you can use 1, `run_statuses`.  And then use a simple any(), all() to find the overall status.
Here is a helpful logic table.
https://stackoverflow.com/a/19389957

Reply:
I am sorry but I don't see how that would work in this case.

Reply:
So, with just one list, can we do the following?
run_statuses = all([True, True, True])  # SUCCESS
run_statuses = any([True, False, True])  # PARTIAL_SUCCESS

Reply:
Sure, but I also need the names of the scanners that failed in order to log the error message.